catalog/spatial/wfs/1.1.0/spatial-wfs-v1_1_0-source/src/main/java/org/codice/ddf/spatial/ogc/wfs/v110/catalog/source/WfsFilterDelegate.java

```diff
@@ -1078,12 +1082,20 @@ private FilterType buildSingleGeospatialFilter(
 
     Envelope envelope = createEnvelopeFromWkt(wkt);
 
+    List<Double> lowerCornerCoordinates = asList(envelope.getMinX(), envelope.getMinY());
```

Comment:
✏️ In an ideal world, I would like a more pure OO solution. For example, a class called LatLonList with a ctor that takes two doubles and the isLatLonOrder flag. The class would then have an asList() method that returns a java.util.List that is constructed in the correct order. The objective is to group all the data (the double values and flag) with the methods (the asList) into one class.

Reply:
I've updated this class to use the strategy pattern. Now `WfsSource` will pass in either a `LatLonCoordinateStrategy` or `LonLatCoordinateStrategy` and the filter delegate does not have any conditional logic for handling the coordinate order.